app/controllers/budgets_controller.rb

```diff
@@ -1,20 +1,26 @@
+# frozen_string_literal: true
+
 class BudgetsController < ApplicationController
   before_action :authenticate_user!
   before_action :set_budget, only: %i[show edit update destroy]
+  before_action :set_category_id, only: %i[index show]
+
+  rescue_from ActiveRecord::RecordNotFound, with: :budget_not_found
 
   def index
     @budgets = current_user.budgets
   end
 
-  def show
-  end
+  def show; end
 
   def new
-    @budget = current_user.budgets.build
+    @budget = Budget.new
+    @categories = Category.all
   end
 
   def create
     @budget = current_user.budgets.build(budget_params)
+    set_month_and_year(@budget)
```

Comment:
- Remove `period`, `month` and `year` from budgets and use only 1 field to store the information of budget month.
- User should be provided with a dropdown which will populate next 12 months with year starting from current month.